chef/cookbooks/provisioner/recipes/bootdisk.rb

```diff
@@ -31,34 +31,35 @@
     basedir = "/dev/disk/by-path"
     dev = nil
     disk_by_path = nil
-    ["-usb-", nil].each do |deviceignore|
-      ::Dir.entries(basedir).sort.each do |path|
-        # Not a symlink?  Not interested.
-        next unless File.symlink?(File.join(basedir, path))
-        # Symlink does not point at a disk?  Also not interested.
-        dev = File.readlink("#{basedir}/#{path}").split("/")[-1]
-        # Prefer devices in a specific order
-        next if (deviceignore and path.include?(deviceignore))
-        disk_by_path = "disk/by-path/#{path}"
-        break if dev =~ /^[hsv]d[a-z]+$/
-        # pci-0000:0b:08.0-cciss-disk0 -> ../../cciss/c0d0
-        break if dev =~ /^c[0-9]+d[0-9]+$/
-        # xen-vbd-51712-part1 -> ../../xvda1
-        break if dev =~ /^xvd[a-z]+$/
-        dev = nil
-        disk_by_path = nil
+    if File.directory?(basedir)
+      ["-usb-", nil].each do |deviceignore|
+        ::Dir.entries(basedir).sort.each do |path|
+          # Not a symlink?  Not interested.
+          next unless File.symlink?(File.join(basedir, path))
+          # Symlink does not point at a disk?  Also not interested.
+          dev = File.readlink("#{basedir}/#{path}").split("/")[-1]
+          # Prefer devices in a specific order
+          next if (deviceignore and path.include?(deviceignore))
```

Comment:
Style/AndOr: Use `&&` instead of `and`.<br>Style/ParenthesesAroundCondition: Don't use parentheses around the condition of an `if`.